src/renderer/wg_engine/tvgWgGeometry.cpp

```diff
@@ -351,11 +351,41 @@ void WgGeometryData::appendStrokeDashed(const WgPolyline* polyline, const Render
 }
 
 
+void WgGeometryData::appendStrokeJoint(WgPoint& v0, WgPoint& v1, WgPoint& v2, StrokeJoin join, float wdt, float limit)
+{
+    WgPoint dir0 = (v1 - v0).normal();
+    WgPoint dir1 = (v2 - v1).normal();
+    WgPoint nrm0 { +dir0.y, -dir0.x };
+    WgPoint nrm1 { +dir1.y, -dir1.x };
+    WgPoint offset0 = nrm0 * wdt;
+    WgPoint offset1 = nrm1 * wdt;
+    if (join == StrokeJoin::Round) {
+        appendCircle(v1, wdt);
+    } else if (join == StrokeJoin::Bevel) {
+        appendRect(v1 - offset0, v1 + offset0, v1 - offset1, v1 + offset1);
+    } else if (join == StrokeJoin::Miter) {
+        WgPoint nrm = (nrm0 + nrm1);
+        if ((!mathZero(nrm.length()) && (!dir0.equal(dir1)))) {
```

Comment:
slightly cheaper method - "2d cross product". we check the directions, ignore the orientation:
`if (!mathZero(dir0.x * dir1.y -  dir0.y * dir1.x)) {`